crates/autopilot/src/infra/blockchain/dto.rs

```diff
@@ -0,0 +1,24 @@
+impl TryFrom<web3::types::Transaction> for crate::domain::settlement::Transaction {
+    type Error = &'static str;
```

Comment:
Should either be `anyhow::Error` if it's intended to be a black box or a dedicated error enum if we intend to match on the error to recover somehow.
Seems like `anyhow::Error` would be the more appropriate choice here.
Same comment below.